src/XliffTasks/Tasks/UpdateXlf.cs

```diff
@@ -43,11 +43,18 @@ protected override void ExecuteCore()
                     {
                         xlfDocument = LoadXlfDocument(xlfPath, language, createIfNonExistent: AllowModification);
                     }
-                    catch (FileNotFoundException ex) when (ex.FileName == xlfPath)
+                    catch (FileNotFoundException fileNotFoundEx) when (fileNotFoundEx.FileName == xlfPath)
                     {
                         Release.Assert(!AllowModification);
                         throw new BuildErrorException($"'{xlfPath}' for '{sourcePath}' does not exist. {HowToUpdate}");
                     }
+                    catch (System.Xml.XmlException xmlEx)
+                    {
+                        throw new BuildErrorException("Unable to load file.", xmlEx)
+                        {
+                            Data = { { BuildErrorException.RelatedFile, xlfPath } }
```

Comment:
I'm not sure if I like this or not. Perhaps I'm being too clever, and should just put the file path in the exception message instead of messing with the `Exception.Data` dictionary.

Reply:
How about just adding an actual RelatedFile property to BuildErrorException? It's basically the same, but easier to read.

Reply:
D'oh! Duh, yes, that would be better. Somehow I latched on to `Exception.Data` and got tunnel vision. :-)